content/blog/2020-05-26-may-20-community-gems.md

```diff
@@ -0,0 +1,93 @@
+
+That will be fine! Having an additional file in the remote that isn't part of
+DVC tracking won't pose any issues. You would only encounter problems if you
+were manually modifying or deleting contents of the remote managed by DVC.
+
+### Q: [Are there limits to how many files DVC can handle? My dataset contains ~100,000 files.](https://discord.com/channels/485586884165107732/563406153334128681/706538115048669274)
+
+We ourselves have stored datasets containing up to 2 million files, so 100,000
+is certainly feasible. Of course, the larger your dataset, the more time data
+transfer operations will take. Luckily, we have a
+[DVC 1.0 contains several data transfer optimizations](https://dvc.org/blog/dvc-3-years-and-1-0-release#data-transfer-optimizations)
+to substantially reduce the time needed to `dvc pull / push / status -c / gc -c`
+for very large datasets.
+
+### Q: [Two developers on my team are doing `dvc push` to the same remote. Should they `dvc pull` first?](https://discord.com/channels/485586884165107732/563406153334128681/704211629075857468)
+
+We encourage users to think of a DVC project like using Git- the same workflow
+applies, except that certain files are pushed to a DVC remote instead of a Git
+remote. In a typical Git flow, developers would `git pull` before
+`git commit & push`. Likewise, `dvc pull` before `dvc push` is a good idea to
+make sure a developer's local workspace is synced with the master branch of a
+project before pushing changes.
+
+With that said, because DVC remotes stores files indexed by `md5`s, there's
+usually a very low probability of a collision (if two developers have two
+different versions of a file, they'll be stored as two separate files in the DVC
+remote- so no merge conflicts). Still, being in the habit of frequent pulling is
+a good practice.
+
+### Q: [What are `*.tmp` files in my DVC remote?](https://discord.com/channels/485586884165107732/563406153334128681/698163554095857745)
+
+Inside your DVC remote, you might see `.tmp` files from incomplete uploads. This
+can happen if a user killed a process like `dvc push`. You can safely remove
+them; for example, if you're using an S3 bucket, `aws s3 rm ... *.tmp` will do
+the trick.
+
+### Q: [I'm using Google Cloud Storage as a DVC remote. I'm getting an error: `ERROR: unexpected error - ('invalid_grant: Bad Request', '{\n "error": "invalid_grant",\n "error_description": "Bad Request"\n}')`. Any ideas?](https://discord.com/channels/485586884165107732/485596304961962003/705131622537756702)
```

Comment:
can we simplify the error message a bit (potentially move json into code block at the beginning of the A):

E.g. if you are getting:

```json
---> error JSON comes here
```

it usually means a permission ...

Reply:
I think we should clarify and expand the answer a little bit here . First I would check the `dvc remote add` and `dvc remote modify` docs - they should have some links to external docs on how to setup Google Cloud CLI and authenticate it. We can at least put those link. @efiop can potentially help with some default flow and/or some external links that describe it.

Reply:
i've expanded the answer, but i'm not positive it's 100% accurate. I don't think DVC tracks the `.json` containing a user's credentials, it sounds like that's managed by GCP CLI. Let me know what you think in the next commit